service/controller/resource/nodepool/create_deployment_uninitialized.go

```diff
@@ -136,11 +148,53 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 		return currentState, nil
 	default:
 		r.Logger.LogCtx(ctx, "level", "debug", "message", "template and parameters unchanged")
+
+		err := r.saveAzureIDsInCR(ctx, virtualMachineScaleSetsClient, virtualMachineScaleSetVMsClient, &azureMachinePool)
+		if err != nil {
```

Comment:
I'm making sure we don't show errors if we couldn't save the data in the CR. We log the error and it will try again on the next reconciliation loop.